dspace-api/src/main/java/org/dspace/util/SolrUpgradeStatistics6.java

```diff
@@ -0,0 +1,371 @@
+import org.dspace.content.service.CommunityService;
+import org.dspace.content.service.ItemService;
+import org.dspace.core.ConfigurationManager;
+import org.dspace.core.Constants;
+import org.dspace.core.Context;
+import org.dspace.services.ConfigurationService;
+import org.dspace.services.factory.DSpaceServicesFactory;
+import org.dspace.statistics.ObjectCount;
+import org.dspace.statistics.SolrLoggerServiceImpl.ResultProcessor;
+import org.dspace.statistics.factory.StatisticsServiceFactory;
+import org.dspace.statistics.service.SolrLoggerService;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.io.IOException;
+import java.net.URL;
+import java.sql.SQLException;
+import java.text.*;
+import java.util.*;
+
+/**
+ * Utility class to export, clear and import Solr indexes.
+ * @author Andrea Schweer schweer@waikato.ac.nz for the LCoNZ Institutional Research Repositories
+ */
+public class SolrUpgradeStatistics6
+{
+        private static final String INDEX_NAME_OPTION = "i";
+        private static final String NUMREC_OPTION = "n";
+        private static final String TYPE_OPTION = "t";
+        private static final String HELP_OPTION = "h";
+        private static final int NUMREC_DEFAULT = 10000;
+        private static final String INDEX_DEFAULT = "statistics";
+        private static final Logger log = Logger.getLogger(SolrUpgradeStatistics6.class);
+        private ConfigurationService configurationService = DSpaceServicesFactory.getInstance().getConfigurationService();
+        private HttpSolrServer server;
+        private int numRec = NUMREC_DEFAULT;
+        private int numProcessed = 0;
+        private enum FIELD{owningColl,owningComm,id,owningItem,scopeId;}
+        private Context context;
+        private Integer type;
+        
+        protected CommunityService communityService = ContentServiceFactory.getInstance().getCommunityService();
+        protected CollectionService collectionService = ContentServiceFactory.getInstance().getCollectionService();
+        protected ItemService itemService = ContentServiceFactory.getInstance().getItemService();
+        protected BitstreamService bitstreamService = ContentServiceFactory.getInstance().getBitstreamService();
+
+        protected SolrLoggerService solrLoggerService = StatisticsServiceFactory.getInstance().getSolrLoggerService();
+
+        public SolrUpgradeStatistics6(String indexName, int numRec, Integer type) {
+                String serverPath = configurationService.getProperty("solr-statistics.server");
+                serverPath = serverPath.replaceAll("statistics$", indexName);
+                System.out.println("Connecting to " + serverPath);
+                server = new HttpSolrServer(serverPath);
+                this.numRec = numRec;
+                this.context = new Context();
+                this.context.setMode(Context.Mode.READ_ONLY);
```

Comment:
Just a small remark, you can pass the `Context.Mode.READ_ONLY` directly to the constructor.